browser/ipfs/ipfs_host_resolver.cc

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+#include <vector>
+
+#include "brave/browser/ipfs/ipfs_host_resolver.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "net/base/host_port_pair.h"
+#include "net/dns/public/dns_protocol.h"
+
+namespace ipfs {
+
+IPFSHostResolver::IPFSHostResolver(
+    network::mojom::NetworkContext* network_context,
+    const std::string& prefix)
+    : prefix_(prefix), network_context_(network_context) {
+  DCHECK(network_context);
+}
+IPFSHostResolver::~IPFSHostResolver() {}
+
+void IPFSHostResolver::Resolve(const net::HostPortPair& host,
+                               const net::NetworkIsolationKey& isolation_key,
+                               net::DnsQueryType dns_query_type,
+                               HostTextResultsCallback callback) {
+  if (!callback)
+    return;
+  if (IsRunning())
+    Reset();
+
+  resolving_host_ = host.host();
+  net::HostPortPair local_host_port(prefix_ + resolving_host_, host.port());
```

Comment:
shouldn't `_dnslink` be appended to the domain name and not the hostname? For instance if you have `www.whatever.com`, you want `_dnslink.whatever.com`, right? Not `_dnslink.www.whatever.com`?

Reply:
I'm looking at https://docs.ipfs.io/concepts/dnslink/#publish-using-a-subdomain and it's a bit confusing because it talks about domains, not hostnames